agent/job_runner.go

```diff
@@ -178,10 +178,19 @@ var _ jobRunner = (*JobRunner)(nil)
 
 // Initializes the job runner
 func NewJobRunner(ctx context.Context, l logger.Logger, apiClient APIClient, conf JobRunnerConfig) (jobRunner, error) {
+	// If the accept response has a token attached, we should use that instead of the Agent Access Token that
+	// our current apiClient is using
+	if conf.Job.Token != "" {
+		clientConf := apiClient.Config()
+		clientConf.Token = conf.Job.Token
+		apiClient = apiClient.New(clientConf)
```

Comment:
Here's where we ensure request headers are propagated to the new client when switching from agent session token to agent job token. The previous `api.NewClient(…)` is on old line 202; I did a bit of refactoring so that we're just mutating/replacing _local vars_ before constructing any structs.